sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/AbstractCompilerCommand.java

```diff
@@ -84,6 +121,128 @@ public void runCommand(ErrorCollector errors) {
+  private void createGraphqlKafkaSources(Path rootDir, Path buildDir, SqrlConfig config) {
+    ResourceResolver resourceResolver = new FileResourceResolver(buildDir);
+
+    Map<String, Optional<String>> scriptFiles = ScriptConfiguration.getFiles(config);
+    Optional<URI> graphqlSchema = scriptFiles.get(ScriptConfiguration.GRAPHQL_KEY)
+            .flatMap(file -> resourceResolver.resolveFile(NamePath.of(file)));
+
+    if (graphqlSchema.isEmpty()) {
+      return;
+    }
+    Path schemaPath = Paths.get(graphqlSchema.get());
+    String schemaName = schemaPath.getFileName().toString().split("\\.")[0];
+
+    String schema = Files.readString(schemaPath);
+    TypeDefinitionRegistry registry = new SchemaParser().parse(schema);
+
+    ObjectTypeDefinition mutationType = (ObjectTypeDefinition) registry
+            .getType("Mutation")
+            .orElse(null);
+    if (mutationType == null) {
+      return;
+    }
+
+    Path schemaRoot = buildDir.resolve(schemaName);
+    Files.createDirectories(schemaRoot);
+    for (graphql.language.FieldDefinition definition : mutationType.getFieldDefinitions()) {
+      String bootstrapServers;
+      if (kafkaStarted) {
+        CLUSTER.createTopic(definition.getName());
+        bootstrapServers = CLUSTER.bootstrapServers();
+      } else {
+        bootstrapServers = "kafka:9092";
+      }
+      Map sourceConfig = createSourceConfig(definition.getName(), bootstrapServers);
+
+      //Check local dir to not overwrite
+      String tableName = definition.getName() + ".table.json";
+
+      if (!Files.exists(rootDir.resolve(schemaName).resolve(tableName))) {
+        Path f = schemaRoot.resolve(tableName);
+        SqrlObjectMapper.INSTANCE.writerWithDefaultPrettyPrinter()
+                .writeValue(f.toFile(), sourceConfig);
+      }
+    }
+  }
+
+
+  private Map createSourceConfig(String tableName, String bootstrapServers) {
+    Map connector = new HashMap();
+    connector.put("name", "kafka");
+    connector.put("bootstrap.servers", bootstrapServers);
+    connector.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+    connector.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+    connector.put(ConsumerConfig.GROUP_ID_CONFIG, "kafka-topic-event-lister");
+    connector.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+    connector.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+    connector.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+    connector.put("topic", tableName);
+
+    Map config = Map.of("type", ExternalDataType.source_and_sink.name(),
+            "canonicalizer", "system",
+            "format", Map.of(
+                    "name", "json"
+            ),
+            "identifier", tableName,
+            "schema", "flexible",
+            "connector", connector
```

Comment:
Why do you need the connector in there? Is that for the server?

Reply:
The database server will also be able to resolve connectors to have the same general architecture as the stream server. 

Since each mutation can resolve a different connector, it makes sense to be here.